probe-rs/src/architecture/xtensa/communication_interface.rs

```diff
@@ -190,7 +190,14 @@ impl XtensaCommunicationInterface {
         self.xdm.halt_on_reset(false);
 
         // TODO: this is only necessary to run code, so this might not be the best place
-        self.write_register_untyped(Register::CurrentPs, 0x40021)?;
+        // Make sure the CPU is in a known state and is able to run code we download.
+        self.write_register({
+            let mut ps = ProgramStatus(0);
```

Comment:
I like this writing a lot! Maybe we can still explore a proper fluent API - which would require us to rewrite bitfields. Def. something for the future, just had the thought :)

Reply:
Yeah I can't say I'm a fan of bitfields.